Created recipe builder and recipe object.

RedFoxity opened the pull request:
Created recipe object with constructor currently accepting name, list of ingredients, time to prepare, price of recipe, and description.

Created recipeBuilder for API caller to interact with to generate instances of recipe.

RedFoxity committed:
Created recipe object with constructor currently accepting name, list of ingredients, time to prepare, price of recipe, and description.

Created recipeBuilder for API caller to interact with to generate instances of recipe.

RedFoxity commented:
Check failed because I'm importing entities.ingredient; which exists in main but not in recipe builder branch.

EricKarpovits commented on src/main/java/entities/recipe.java:
```diff
@@ -0,0 +1,32 @@
+        this.dict.put("IngredientList",ing);
+        this.dict.put("TimeToPrepare",time);
+        this.dict.put("Cost",price);
+        this.dict.put("Description",info);
```
Please have your code formatted so that it is consistent. For example '"Name", name' has a space after the comma but '"TimeToPrepare",time);' doesn't. I think you can let IntelliJ do it by doing Ctrl+Alt+L or under Code and Reformat Code.

EricKarpovits commented on src/main/java/useCases/recipeBuilder.java:
```diff
@@ -0,0 +1,18 @@
+     */
+    public static recipe createRecipe(String name,ArrayList<ingredient> ingList, int time, int price, String info){
+        recipe n = new recipe(name, ingList, time, price, info);
+        return n;
```
Consider using a more meaningful variable name than n or consider doing 'return new recipe(name, ingList, time, price, info);'

Further, the abbreviation ingr might be better than ing so that we don't have any ambiguities. [Ingr](https://www.abbreviations.com/term/2201113)

EricKarpovits replied:
Unless everyone is good with using ing.

fonuxxine committed:
Update attribute names

fonuxxine committed:
Update attribute names and styling

fonuxxine committed:
Merge branch 'main' of https://github.com/CSC207-2022F-UofT/course-project-recipez into Recipe-Builder

RedFoxity committed:
Merge branch 'main' of https://github.com/CSC207-2022F-UofT/course-project-recipez into Recipe-Builder

EricKarpovits commented on src/main/java/entities/Recipe2.java:
```diff
@@ -0,0 +1,32 @@
+package entities;
```
I think Recipie2.java is not a great name for the file.

EricKarpovits commented on src/main/java/entities/Recipe2.java:
```diff
@@ -0,0 +1,32 @@
+     * recipe object to be instantiated by recipeBuilder class.
+     * Information regarding a recipe is then collected in a dictionary within the instance
+     */
+     private Dictionary<String, Object> dict = new Hashtable();
```
This shouldn't be declared like this. The instance variable should be created in the constructor of the Recipe Class.

EricKarpovits commented on src/main/java/entities/Recipe2.java:
```diff
@@ -0,0 +1,32 @@
+     * Single class constructor requires name of recipe, an ArrayList of required ingredients, time to prepare
+     * cost of preparation, and other info such as instructions/description.
+     */
+     public recipe(String name, ArrayList<Ingredient> ing, int time, int price, String info){
```
Is this the constructor if so It should be capitalized R.

EricKarpovits commented on src/main/java/entities/Recipe2.java:
```diff
@@ -0,0 +1,32 @@
+     * Returns a dictionary with keys, "Name", "IngredientList", "TimeToPrepare", "Cost", and "Description".
+     */
+    public Dictionary<String, Object> getRecipeInfo(){
+        return dict;
```
consider using this.dict as it is the best convention.

fonuxxine committed:
Implement uses cases for Fridge and Recipe

RedFoxity committed:
- Fixed bugs/errors (Syntax errors, incorrect references, etc..)
- Added Factories and interfaces for Ingredients
- Added CommonIngredient that implements Ingredient
- Added CommonRecipeFactory that implements RecipeFactory
- Changed RecipeBuilder to implement RecipeInputBoundary
- Changed RecipeBuilder to a String[] for createRecipe method
- Added APIOutputFormatter which implements RecipePresenter
- Added 2 unit tests for adding ingredients to fridge and creating a recipe

EricKarpovits commented on src/main/java/entities/CommonRecipeFactory.java:
```diff
@@ -0,0 +1,8 @@
+package entities;
+
+public class CommonRecipeFactory implements RecipeFactory{
```
Please add a class docstring

RedFoxity committed:
Merge remote-tracking branch 'origin/Recipe-Builder' into Recipe-Builder

fonuxxine committed:
Remove raw use of Dictionary object

fonuxxine committed:
Add Recipe Builder Controller, Response Model, Request Model, Interactor, Input Boundary

RedFoxity committed:
Merge remote-tracking branch 'origin/Recipe-Builder' into Recipe-Builder

fonuxxine committed:
Add docstring to classes, interface, methods

RedFoxity committed:
Merge remote-tracking branch 'origin/Recipe-Builder' into Recipe-Builder

RedFoxity committed:
- Added Gson dependency in gradle.build (Allows us to use google Json parser)
- Reformatted Recipe inputs and variables (Recipes are now composed of it's name, link, and image)
- Completely redid the parsing algorithm in RecipeBuilderResponseModel (Now accepts "infinite" recipe input
- Changed CreateRecipeAndTestName unit test to reflect changes

RedFoxity committed:
- Added Gson dependency in gradle.build (Allows us to use google Json parser)
- Reformatted Recipe inputs and variables (Recipes are now composed of it's name, link, and image)
- Completely redid the parsing algorithm in RecipeBuilderResponseModel (Now accepts "infinite" recipe input
- Changed CreateRecipeAndTestName unit test to reflect changes

EricKarpovits reviewed (approved):
Looks great now. Make sure to resolve the merge conflicts.

rickychen175 reviewed (approved):
The RecipeBuilder looks good and follows design principles well.

EricKarpovits committed:
Fixed merge conflicts

EricKarpovits committed:
Fixed the unit tests

EricKarpovits committed:
Fixed the unit tests again

Shaffaan reviewed (approved):
The code looks good and really nice that it works for multiple recipe objects

EricKarpovits committed:
Added Serializable

EricKarpovits committed:
Fixed the commonRecipeFactory's docstrings
